base/base.py

```diff
@@ -17,8 +21,51 @@ def read_app_data(self, file_name, file_path=None):
         	Returns:
         	    Parsed data in dictionary
         '''
-        if td_dir is None:
-            td_dir = self.test["case_td_dir"]
+        if file_name is None or class_name is None:
+            print('Cannot get the test data file name') 
```

Comment:
Instead of checking like this, better practice is to make them mandatory arguments. i.e, don't give the default value "None"